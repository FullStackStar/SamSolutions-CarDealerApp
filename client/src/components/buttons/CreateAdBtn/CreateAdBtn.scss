@import "../../../styles/variables";

.button-create {
  width: 9em;
  background-color: $secondary_buttons_color;
  color: $header_footer_color;
  box-shadow: 0 0 3px $box_shadow_color;
  cursor: pointer;
}
.button-create:hover {
  background-color: $secondary_buttons_hover_color;
}
.button-create__container {
  position: absolute;
  right: 4em;
  top: 27em;
  margin-top: 1em;
}
