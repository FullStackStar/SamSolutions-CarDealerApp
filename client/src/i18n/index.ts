import { format } from "path";

export { default as I18nProvider } from "./provider";
export { LOCALES } from "./locales";
