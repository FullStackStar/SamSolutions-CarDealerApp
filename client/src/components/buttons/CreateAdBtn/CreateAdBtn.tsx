import React from "react";

import "./CreateAdBtn.scss";
import "../SearchButton/SearchButton";

class CreateAdBtn extends React.Component<any, {}> {
  render() {
    return (
      <div className="button-create__container absolute" data-testid="createAdBtn">
        <button
          className="button button-create"
          onClick={this.props.handleClick}
        >
          {this.props.content}
        </button>
      </div>
    );
  }
}

export default CreateAdBtn;
