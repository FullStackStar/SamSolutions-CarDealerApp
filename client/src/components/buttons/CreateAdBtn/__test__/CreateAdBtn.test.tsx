import React from 'react';
import ReactDOM from 'react-dom';
import CreateAdBtn from '../CreateAdBtn';
import { I18nProvider, LOCALES } from "../../../../i18n";

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom";

import renderer from 'react-test-renderer';

afterEach(cleanup);

it("CreateAdBtn component renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(<I18nProvider locale={LOCALES.ENGLISH}><CreateAdBtn /></I18nProvider>, div)
});

it("Create ad button renders correctly", () => {
    const {getByTestId} = render(<I18nProvider locale={LOCALES.ENGLISH}><CreateAdBtn content="Create new ad"/></I18nProvider>);
    expect(getByTestId("createAdBtn")).toHaveTextContent("Create new ad")
});
it("Create ad button renders correctly", () => {
    const {getByTestId} = render(<I18nProvider locale={LOCALES.ENGLISH}><CreateAdBtn content="Save"/></I18nProvider>);
    expect(getByTestId("createAdBtn")).toHaveTextContent("Save") 
});

it("Create ad button matches snapshot 1", () => {
    const tree = renderer.create(<I18nProvider locale={LOCALES.ENGLISH}><CreateAdBtn content="Create new ad"/></I18nProvider>).toJSON();
    expect(tree).toMatchSnapshot();
});
it("Create ad button matches snapshot 2", () => {
    const tree = renderer.create(<I18nProvider locale={LOCALES.ENGLISH}><CreateAdBtn content="Save"/></I18nProvider>).toJSON();
    expect(tree).toMatchSnapshot();
})
