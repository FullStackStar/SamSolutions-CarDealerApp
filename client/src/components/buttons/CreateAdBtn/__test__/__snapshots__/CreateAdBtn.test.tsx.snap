// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create ad button matches snapshot 1`] = `
<div
  className="button-create__container absolute"
  data-testid="createAdBtn"
>
  <button
    className="button button-create"
  >
    Create new ad
  </button>
</div>
`;

exports[`Create ad button matches snapshot 1 1`] = `
<div
  className="button-create__container absolute"
  data-testid="createAdBtn"
>
  <button
    className="button button-create"
  >
    Create new ad
  </button>
</div>
`;

exports[`Create ad button matches snapshot 2 1`] = `
<div
  className="button-create__container absolute"
  data-testid="createAdBtn"
>
  <button
    className="button button-create"
  >
    Save
  </button>
</div>
`;
